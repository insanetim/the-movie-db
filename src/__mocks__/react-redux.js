export const dispatch = jest.fn()

export const useDispatch = jest.fn(() => dispatch)
export const useSelector = jest.fn(fn => fn())
