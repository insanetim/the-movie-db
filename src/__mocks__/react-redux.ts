const dispatch = jest.fn()
const getState = jest.fn()

const useDispatch = jest.fn(() => dispatch)
const useSelector = jest.fn(fn => fn())

export { dispatch, getState, useDispatch, useSelector }
