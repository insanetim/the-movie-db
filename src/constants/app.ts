export const APP_NAME = 'THE MOVIE DB'
export const API_URL = 'https://api.themoviedb.org/3/'
export const ACCESS_TOKEN_AUTH =
  'eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI1N2RiYjI4M2M4MDBjY2QwYmM5YTczNTJjN2Y1ZmEzYiIsInN1YiI6IjYzMmM4YThkZTNmYTJmMDA3ZmQ0ZjY5OSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.nMVY53gL3_w8JAHI-N_uKQyXpvT9TsZ4_p8nhJSnAdQ'
export enum NOTIFICATION_TYPE {
  ERROR = 'error',
  SUCCESS = 'success'
}
export const NOTIFICATION_DURATION = 2.5
