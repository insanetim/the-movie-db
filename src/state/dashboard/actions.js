import * as types from './types'

export const fetchTrending = page => ({
  type: types.FETCH_TRENDING,
  payload: page
})

export const fetchTrendingRequest = () => ({
  type: types.FETCH_TRENDING_REQUEST
})

export const fetchTrendingSuccess = movies => ({
  type: types.FETCH_TRENDING_SUCCESS,
  payload: movies
})

export const fetchTrendingError = error => ({
  type: types.FETCH_TRENDING_ERROR,
  payload: error
})

export const setTrendingPage = page => ({
  type: types.SET_TRENDING_PAGE,
  payload: page
})

export const fetchSearch = ({ page, query }) => ({
  type: types.FETCH_SEARCH,
  payload: { page, query }
})

export const fetchSearchRequest = () => ({
  type: types.FETCH_SEARCH_REQUEST
})

export const fetchSearchSuccess = movies => ({
  type: types.FETCH_SEARCH_SUCCESS,
  payload: movies
})

export const fetchSearchError = error => ({
  type: types.FETCH_SEARCH_ERROR,
  payload: error
})

export const setSearchPage = page => ({
  type: types.SET_SEARCH_PAGE,
  payload: page
})
