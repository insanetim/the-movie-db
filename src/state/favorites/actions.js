import * as types from './types'

export const fetchFavorites = page => ({
  type: types.FETCH_FAVORITES,
  payload: page
})

export const fetchFavoritesRequest = page => ({
  type: types.FETCH_FAVORITES_REQUEST,
  payload: page
})

export const fetchFavoritesSuccess = movies => ({
  type: types.FETCH_FAVORITES_SUCCESS,
  payload: movies
})

export const fetchFavoritesFailure = error => ({
  type: types.FETCH_FAVORITES_FAILURE,
  payload: error
})

export const setFavoritesPage = page => ({
  type: types.SET_FAVORITES_PAGE,
  payload: page
})
