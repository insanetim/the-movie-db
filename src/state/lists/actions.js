import * as types from './types'

export const fetchLists = ({ page, callback }) => ({
  type: types.FETCH_LISTS,
  payload: { page, callback }
})

export const fetchListsRequest = page => ({
  type: types.FETCH_LISTS_REQUEST,
  payload: page
})

export const fetchListsSuccess = movies => ({
  type: types.FETCH_LISTS_SUCCESS,
  payload: movies
})

export const fetchListsFailure = error => ({
  type: types.FETCH_LISTS_FAILURE,
  payload: error
})

export const setListsPage = page => ({
  type: types.SET_LISTS_PAGE,
  payload: page
})

export const fetchList = listId => ({
  type: types.FETCH_LIST,
  payload: listId
})

export const fetchListRequest = () => ({
  type: types.FETCH_LIST_REQUEST
})

export const fetchListSuccess = movies => ({
  type: types.FETCH_LIST_SUCCESS,
  payload: movies
})

export const fetchListFailure = error => ({
  type: types.FETCH_LIST_FAILURE,
  payload: error
})

export const createList = ({ value, callback }) => ({
  type: types.CREATE_LIST,
  payload: { value, callback }
})

export const addToList = ({ listId, movieId }) => ({
  type: types.ADD_TO_LIST,
  payload: { listId, movieId }
})

export const removeFromList = ({ listId, movieId }) => ({
  type: types.REMOVE_FROM_LIST,
  payload: { listId, movieId }
})

export const deleteList = ({ listId, callback }) => ({
  type: types.DELETE_LIST,
  payload: { listId, callback }
})
