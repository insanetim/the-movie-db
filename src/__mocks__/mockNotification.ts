import { NOTIFICATION_DURATION, NOTIFICATION_TYPE } from 'src/constants'
import { Notification } from 'src/store/features/app'

const mockNotification: Notification = {
  duration: NOTIFICATION_DURATION,
  id: 'test/id',
  message: 'test/message',
  type: NOTIFICATION_TYPE.SUCCESS,
}

export default mockNotification
