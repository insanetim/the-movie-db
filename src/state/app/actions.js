import { v4 as uuid } from 'uuid'

import { NOTIFICATION_DURATION, NOTIFICATION_TYPE } from 'src/constants'

import * as types from './types'

export const showModal = payload => ({
  type: types.SHOW_MODAL,
  payload
})

export const hideModal = () => ({
  type: types.HIDE_MODAL
})

export const showNotification = ({
  id = uuid(),
  messageType = NOTIFICATION_TYPE.SUCCESS,
  messageText,
  duration = NOTIFICATION_DURATION
}) => ({
  type: types.SHOW_NOTIFICATION,
  payload: {
    id,
    messageType,
    messageText,
    duration
  }
})

export const hideNotification = payload => ({
  type: types.HIDE_NOTIFICATION,
  payload
})
