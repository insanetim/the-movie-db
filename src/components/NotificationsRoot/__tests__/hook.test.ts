import { act, renderHook } from '@testing-library/react'
import mockNotification from 'src/__mocks__/mockNotification'
import { dispatch } from 'src/__mocks__/react-redux'
import { hideNotification as hideNotificationAction } from 'src/store/app/actions'

import useContainer from '../hook'

jest.mock('src/store/app/selectors', () => ({
  notificationsSelector: jest.fn(() => [mockNotification])
}))

describe('NotificationsRoot useContainer hook', () => {
  it('should match snapshot', () => {
    const { result } = renderHook(useContainer)

    expect(result.current).toMatchSnapshot()
  })

  it('should check `hideNotification` method', () => {
    const { result } = renderHook(useContainer)

    act(() => {
      result.current.hideNotification('test/id')
    })

    expect(dispatch).toHaveBeenCalledWith(hideNotificationAction('test/id'))
  })
})
