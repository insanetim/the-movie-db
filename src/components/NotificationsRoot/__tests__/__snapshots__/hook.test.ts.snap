// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsRoot useContainer hook should match snapshot 1`] = `
{
  "hideNotification": [Function],
  "notifications": [
    {
      "duration": 2.5,
      "id": "nanoid",
      "messageText": "test/message",
      "messageType": "success",
    },
  ],
}
`;
